spec/architecture-decision-records/1681149688-remove-d1-completely.md

```diff
@@ -0,0 +1,122 @@
+# stop using d1 with `@web3-storage/access-api`
+
+<!--
+This is a minimal template. Feel free to add more sections as needed.
+
+Please review also the Design Doc template and add any relevant sections to your ADR:
+https://www.notion.so/pl-strflt/Writing-a-Design-Doc-aa6034be43c2434ba88a2fd844516e94
+-->
+
+> Status: PROPOSED
+
+Authors
+* @gobengo
+* @travis
+
+<!--
+PROPOSED, ACCEPTED, REJECTED, DEPRECATED, SUPERSEDED BY {link-to-ADR}
+-->
+
+## Context
+
+<!--
+What is the issue that we're seeing that motivates this decision or change?
+-->
+
+### Prior Documents
+
+* [`@web3-storage/access-api` data storage](https://hackmd.io/f5PiUFarQJiq6L0K7xaLQA)
+
+### Reasons we need to stop using D1
+
+1. storage limitations are a ticking time bomb in production
```

Comment:
* It's in public alpha and Cloudflare says it shouldn't be used for production products
* It has a 100Mb database size limit
* The size limit caps w3up capacity
* Provide rough estimate of the current capacity before hitting the constraint imposed by D1 - approx `O(spaces) = {d1SizeLimit} / ( {cidSize=32bytes} * {cidsPerSpace} )`